lib/usb/usb_standard.c

```diff
@@ -247,19 +248,30 @@ static int usb_standard_set_configuration(usbd_device *usbd_dev,
 					  struct usb_setup_data *req,
 					  uint8_t **buf, uint16_t *len)
 {
-	int i;
+	uint8_t i;
```

Comment:
then just use "unsigned" if you must.  or "uint8_least_t" or something that will let it use a bigger variable if needs to.  we're on a 32bit platform.